docs: Update abs, filesha1, filesha256, and filesha512 functions (#35599)
hashicorp/terraform: website/docs/language/functions/filesha1.mdx

---
page_title: filesha1 - Functions - Configuration Language
description: The filesha1 function computes the SHA1 hash of the contents of a given file and encodes it as hex.
---

# `filesha1` function reference
This topic provides reference information about the `filesha1` function, which calculates the SHA-1 hash of a file's contents. 

## Introduction

The `filesha1` is a variant of [`sha1`](/terraform/language/functions/sha1) that hashes the contents of a given file rather than a literal string. 

> Use the `filesha1` function instead of wrapping the `file` function in a `sha1` function, for example `sha1(file(filename))`, because [`file`](/terraform/language/functions/file) accepts only UTF-8 text. As a result, you cannot use `sha1(file(filename))`  to create hashes for binary files.

Reviewer: ```suggestion
Use the `filesha1` function instead of wrapping the `file` function in a `sha1` function, for example `sha1(file(filename))`, because [`file`](/terraform/language/functions/file) accepts only UTF-8 text. As a result, you cannot use `sha1(file(filename))` to create hashes for binary files.
```

---
page_title: filesha1 - Functions - Configuration Language
description: he filesha1 function computes the SHA1 hash of the contents of a given file and encodes it as hex.
---

# `filesha1` function reference
This topic provides reference information about the `filesha1` function, which calculates the SHA-1 hash of a file's contents. 

## Introduction

The `filesha1` is a variant of [`sha1`](/terraform/language/functions/sha1) that hashes the contents of a given file rather than a literal string. 

> This is similar to `sha1(file(filename))`, but
> because [`file`](/terraform/language/functions/file) accepts only UTF-8 text it cannot be used to
> create hashes for binary files.

Reviewer 2: ```suggestion
Use the `filesha1` function instead of wrapping the `file` function in a `sha1` function, for example `sha1(file(filename))`, because [`file`](/terraform/language/functions/file) accepts only UTF-8 text. As a result, you cannot use `sha1(file(filename))`  to create hashes for binary files.
```
This is one falls under the 'don't make assumptions' category. Making this type of comparison assumes that the reader knows (even though they probably do) about `sha1(file(filename))`.

> !> **Security warning**: Collision attacks have been successfully performed against this hashing function. Before using this function for anything security-sensitive, review relevant literature to understand the security implications.

Reviewer 2: ```suggestion
!> **Security warning**: This hashing function is susceptible to collision attacks. Before using this function for anything security-sensitive, review relevant literature to understand the security implications.
```
This is sort of the mid-level active voice

## Syntax

Use the filesha1 function with the following syntax:

```hcl
filesha1(path) 
```

> The `path` is the file path (relative or absolute) to the file whose SHA-1 hash you want to compute.

Reviewer 2: ```suggestion
The `path` is the relative or absolute file path to the file whose SHA-1 hash you want to compute.
```

In the following example, the function returns the SHA-1 value.

```hcl
> filesha1("example.txt")
d3486ae9136e7856bc42212385ea797094475802
```

## Example use case

> In the following example the `filesha1` function computes the SHA-1 hash of the file `example.txt` located in the current module's directory. The result will be a 40-character hexadecimal string representing the SHA-1 hash.

Reviewer 2: ```suggestion
In the following example the `filesha1` function computes the SHA-1 hash of the file `example.txt` located in the current module's directory. The result is a 40-character hexadecimal string representing the SHA-1 hash.
```

---
page_title: filesha1 - Functions - Configuration Language
description: he filesha1 function computes the SHA1 hash of the contents of a given file and encodes it as hex.
---

# `filesha1` function reference
This topic provides reference information about the `filesha1` function, which calculates the SHA-1 hash of a file's contents. 

## Introduction

The `filesha1` is a variant of [`sha1`](/terraform/language/functions/sha1) that hashes the contents of a given file rather than a literal string. 

This is similar to `sha1(file(filename))`, but
because [`file`](/terraform/language/functions/file) accepts only UTF-8 text it cannot be used to
create hashes for binary files.

!> **Security warning**: Collision attacks have been successfully performed against this hashing function. Before using this function for anything security-sensitive, review relevant literature to understand the security implications.

## Syntax

Use the filesha1 function with the following syntax:

```hcl
filesha1(path) 
```

The `path` is the file path (relative or absolute) to the file whose SHA-1 hash you want to compute.

In the following example, the function returns the SHA-1 value.

```hcl

> > filesha1("example.txt")

Reviewer: ```suggestion
$ filesha1("example.txt")
```
